drivers/entropy/entropy_gecko_se.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021 Safran Passenger Innovations Germany GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_semailbox
+
+ #include <drivers/entropy.h>
+ #include <soc.h>
+ #include "em_cmu.h"
+ #include "sl_se_manager.h"
+ #include "sl_se_manager_entropy.h"
+
+
+static int entropy_gecko_se_get_entropy(const struct device *dev,
+					uint8_t *buffer,
+					uint16_t length)
+{
+	ARG_UNUSED(dev);
+
+	int err = 0;
+	sl_status_t status;
+	sl_se_command_context_t cmd_ctx;
+
+	status = sl_se_init_command_context(&cmd_ctx);
+
+	if (status == SL_STATUS_OK) {
+		status = sl_se_get_random(&cmd_ctx, buffer, length);
+
+		if (status != SL_STATUS_OK) {
+			err = -EIO;
+		}
+
+		sl_se_deinit_command_context(&cmd_ctx);
+	} else {
+		err = -EIO;
+	}
+
+	return err;
+}
+
+static int entropy_gecko_se_get_entropy_isr(const struct device *dev,
+					    uint8_t *buf,
+					    uint16_t len, uint32_t flags)
+{
+	return -ENOTSUP;
+}
+
+static int entropy_gecko_se_init(const struct device *dev)
+{
+	if (sl_se_init()) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static struct entropy_driver_api entropy_gecko_se_api_funcs = {
+	.get_entropy = entropy_gecko_se_get_entropy,
+	.get_entropy_isr = entropy_gecko_se_get_entropy_isr
+};
+
+DEVICE_DT_INST_DEFINE(0,
```

Comment:
I think it would be a bit cleaner from the user perspective to still use DT_INST_FOREACH_STATUS_OKAY. This will allow the user to disable the devicetree node to prevent allocation of the device, which follows the general Zephyr behavior: drivers are enabled with Kconfig, devices are allocated if and only if devicetree nodes are enabled.

Reply:
It makes sense indeed. I did the modification to use DT_INST_FOREACH_STATUS_OKAY instead.